File: LeetCode-March22/Simplify Path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        cur = ""
        
        for c in path + "/":
            if c == "/":
                if cur == "..": 
                    if stack: stack.pop()
                elif cur != "" and cur != ".":
                    stack.append(cur)
                cur = ""
                    
            else:
                cur += c
        return "/" + "/".join(stack)      

File: Leetcode April 2023/Simplify Path.java
class Solution {
    public String simplifyPath(String path) {
       Stack<String> stack = new Stack<>(); 
        String[] directories = path.split("/"); 
        for (String dir : directories) { 
            if (dir.equals(".") || dir.isEmpty()) {
                continue;
            } else if (dir.equals("..")) { 
                if (!stack.isEmpty()) { 
                    stack.pop();
                }
            } else { 
                stack.push(dir);
            }
        }
        return "/" + String.join("/", stack); 
    }
}
